plugins/c8y_firmware_plugin/src/firmware_manager.rs

```diff
@@ -549,38 +570,26 @@ enum ActiveOperationState {
     Executing,
 }
 
-struct PersistentStore;
-impl PersistentStore {
-    fn get_dir_path() -> PathBuf {
-        PathBuf::from(FIRMWARE_OPERATION_DIR_PATH)
+#[derive(Debug, PartialEq, Eq, Clone)]
+struct PersistentStore {
+    persistent_dir: PathBuf,
+}
+
+impl From<PathBuf> for PersistentStore {
+    fn from(path: PathBuf) -> Self {
+        Self {
+            persistent_dir: path,
+        }
     }
+}
 
-    fn get_file_path(op_id: &str) -> PathBuf {
-        PathBuf::from(FIRMWARE_OPERATION_DIR_PATH).join(op_id)
+impl PersistentStore {
+    fn get_dir_path(&self) -> PathBuf {
+        self.persistent_dir.join(PERSISTENT_STORE_DIR_NAME)
     }
 
-    // TODO: Candidate to move to file.rs
-    fn has_expected_permission(op_id: &str) -> Result<(), FirmwareManagementError> {
-        let path = Self::get_file_path(op_id);
-
-        let metadata = get_metadata(path.as_path())?;
-        let file_uid = metadata.uid();
-        let file_gid = metadata.gid();
-        let tedge_uid = get_uid_by_name("tedge")?;
-        let tedge_gid = get_gid_by_name("tedge")?;
-        let root_uid = get_uid_by_name("root")?;
-        let root_gid = get_gid_by_name("root")?;
-
-        if (file_uid == tedge_uid || file_uid == root_uid)
-            && (file_gid == tedge_gid || file_gid == root_gid)
-            && format!("{:o}", metadata.permissions().mode()).contains("644")
-        {
-            Ok(())
-        } else {
-            Err(FirmwareManagementError::InvalidFilePermission {
-                id: op_id.to_string(),
-            })
-        }
+    fn get_file_path(&self, op_id: &str) -> PathBuf {
```

Comment:
```suggestion
    fn get_operation_file_path(&self, op_id: &str) -> PathBuf {
```
To be more contextual.